### PR TITLE
adds support for tables and queue - aspire

### PR DESCRIPTION
**Pull request by vhvb1989**
fix: https://github.com/Azure/azure-dev/issues/2982

### CODE REVIEW

File: cli/azd/resources/apphost/templates/resources.bicept [hunk 1]
```diff
@@ -225,9 +225,15 @@ output SERVICE_BINDING_{{alphaSnakeUpper $name}}_ENDPOINT string = {{bicepName $
...
-{{range $cname := $value.Containers -}}
+{{range $cname := $value.Blobs -}}
 output SERVICE_BINDING_{{alphaSnakeUpper $cname}}_ENDPOINT string = {{bicepName $name}}.properties.primaryEndpoints.blob
 {{end -}}
+{{range $cname := $value.Tables -}}
+output SERVICE_BINDING_{{alphaSnakeUpper $cname}}_ENDPOINT string = {{bicepName $name}}.properties.primaryEndpoints.table
+{{end -}}
+{{range $cname := $value.Queues -}}
+output SERVICE_BINDING_{{alphaSnakeUpper $cname}}_ENDPOINT string = {{bicepName $name}}.properties.primaryEndpoints.queue
+{{end -}}
```
Comment (wbreza) [inline cli/azd/resources/apphost/templates/resources.bicept:236]: Do these output names need another component so blob/table/queue do not conflict and generate the same names?

File: cli/azd/resources/apphost/templates/main.bicept [hunk 2]
```diff
@@ -45,7 +45,13 @@ output SERVICE_BINDING_{{alphaSnakeUpper $name}}_ENDPOINT string = resources.out
...
 {{end -}}
 {{range $name, $value := .StorageAccounts -}}
-{{range $cname := $value.Containers -}}
+{{range $cname := $value.Blobs -}}
+output SERVICE_BINDING_{{alphaSnakeUpper $cname}}_ENDPOINT string = resources.outputs.SERVICE_BINDING_{{alphaSnakeUpper $cname}}_ENDPOINT
+{{end -}}
+{{range $cname := $value.Tables -}}
+output SERVICE_BINDING_{{alphaSnakeUpper $cname}}_ENDPOINT string = resources.outputs.SERVICE_BINDING_{{alphaSnakeUpper $cname}}_ENDPOINT
+{{end -}}
+{{range $cname := $value.Queues -}}
```
Comment (wbreza) [inline cli/azd/resources/apphost/templates/main.bicept:54]: Do these output names need another component so blob/table/queue do not conflict and generate the same names?

Comment (ellismg) [reply cli/azd/resources/apphost/templates/main.bicept:54]: I don't think so - the `$cname` value here is bound to the name inside the `Tables` slice on the storage account object (note the use of `range` here).  That should correspond to the name of the resource in the aspire manifest, which should be unique across all the names.

**Review by ellismg — CHANGES_REQUESTED** [review]
I think we also need to create role assignments for RBAC to work:

Here's what they do in Asprie today for their F5 experience:

https://github.com/dotnet/aspire/blob/v8.0.0-preview.1.23557.2/src/Aspire.Hosting.Azure.Provisioning/Provisioners/StorageProvisioner.cs#L93-L103

We should probably just create role assignments for the services you use, not all of them, as Aspire does today.

File: cli/azd/resources/apphost/templates/resources.bicept (hunk 1 above)
Comment (vhvb1989) [reply cli/azd/resources/apphost/templates/resources.bicept:263]: aspire make this to be unique

**Review by ellismg — APPROVED** [review]
LGTM!

**Pull request merged**